### PR TITLE
Update artemis version

### PR DESCRIPTION
**Pull request by BeshoyMelika**

### CODE REVIEW

File: example/lib/APIs.dart [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import 'package:mmd_ecommerce_fl_lib/mmd_ecommerce.dart';
 
 callBrandsApi() {
-  GeneralApiManager.brandsApi((List<Brand> brands) {}, () {});
+  GeneralApiManager.brandsApi((List<Brands$Query$Brand> brands) {}, () {});
```
Comment (MinaSamirCap) [inline example/lib/APIs.dart]: find a mapping way to map the names

File: pubspec.yaml [hunk 2]
```diff
@@ -11,13 +12,16 @@ dependencies:
...
-  gql: ^0.12.0
-  meta: '>=1.0.0 <2.0.0' # only if you have non nullable fields
+  equatable: ^1.2.5
+  gql: ">=0.7.3 <1.0.0" # update according to DOCS https://pub.dev/packages/artemis/versions/6.12.3-beta.1#installation
+  meta: ">=1.0.0 <2.0.0" # only if you have non nullable fields
   path_provider: ^1.5.1
   graphql_flutter: ^3.1.0
   analyzer: ^0.39.2+1
-  artemis: ^2.1.4
+  artemis: ^6.12.3-beta.1
```
Comment (MinaSamirCap) [inline pubspec.yaml]: add comments above the library

File: pubspec.yaml [hunk 3]
```diff
@@ -11,13 +12,16 @@ dependencies:
   flutter:
     sdk: flutter
   json_serializable: ^3.2.3
-  equatable: ^1.0.2
-  gql: ^0.12.0
-  meta: '>=1.0.0 <2.0.0' # only if you have non nullable fields
+  equatable: ^1.2.5
+  gql: ">=0.7.3 <1.0.0" # update according to DOCS https://pub.dev/packages/artemis/versions/6.12.3-beta.1#installation
+  meta: ">=1.0.0 <2.0.0" # only if you have non nullable fields
   path_provider: ^1.5.1
```
Comment (MinaSamirCap) [inline pubspec.yaml]: show me the exception of path_provider package

File: lib/common_models/common_models.dart [hunk 4]
```diff
@@ -1,91 +1,91 @@
...
-    as P;
-import 'package:mmd_ecommerce_fl_lib/generatedql/cart/graphql_api.dart' as C;
-import 'package:mmd_ecommerce_fl_lib/generatedql/categories/graphql_api.dart'
-    as CT;
-import 'package:mmd_ecommerce_fl_lib/generatedql/ratings/graphql_api.dart' as R;
-import 'package:mmd_ecommerce_fl_lib/generatedql/related_product/graphql_api.dart'
-    as RELATED;
-import 'package:mmd_ecommerce_fl_lib/generatedql/order_products/graphql_api.dart'
-    as O;
+// import 'package:mmd_ecommerce_fl_lib/generatedql/auth/graphql_api.dart' as A;
```
Comment (MinaSamirCap) [inline lib/common_models/common_models.dart]: if you did not find a map way you will write by your self ..

**Review by MinaSamirCap — CHANGES_REQUESTED** [review]

File: pubspec.yaml (hunk 3 above)
Comment (BeshoyMelika) [reply pubspec.yaml]: fixed after update path_provider 1.6.18

File: example/lib/APIs.dart (hunk 1 above)
Comment (BeshoyMelika) [reply example/lib/APIs.dart]: i open a new issue https://github.com/comigor/artemis/issues/214
and we can return a map<String,String> from api manger and map in example

File: example/lib/apis/address_apis.dart [hunk 5]
```diff
@@ -1,8 +1,8 @@
 import 'package:mmd_ecommerce_fl_lib/mmd_ecommerce.dart';
 
 getAllAddress() async {
-  await AddressApiManager.addressesApi(5, 1,
-      (UserAddressPaginator addressPaginator) {}, (ApiErrorModel error) {});
+  await AddressApiManager.addressesApi(
+      5, 1, (AddressesModel addressPaginator) {}, (ApiErrorModel error) {});
```
Comment (MinaSamirCap) [inline example/lib/apis/address_apis.dart]: AddressesWrapper

File: example/lib/apis/cart_api.dart [hunk 6]
```diff
@@ -16,5 +16,6 @@ addToCard() async {
 }
 
 getCard() async {
-  await CartApiManager.cartDataApi((Cart cart) {}, (ApiErrorModel error) {});
+  await CartApiManager.cartDataApi(
```
Comment (MinaSamirCap) [inline example/lib/apis/cart_api.dart]: cartApi

File: example/lib/apis/general_api.dart [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
 
 callBrandsApi() {
-  GeneralApiManager.brandsApi((List<Brand> brands) {}, () {});
+  GeneralApiManager.brandsApi((BrandModel brands) {}, () {});
 }
 
 sliderApi() async {
   await GeneralApiManager.sliderApi(
-      (List<Slider> list) {}, (ApiErrorModel error) {});
+      (SlidersModel sliderModel) {}, (ApiErrorModel error) {});
```
Comment (MinaSamirCap) [inline example/lib/apis/general_api.dart]: rename sliders model

Comment (MinaSamirCap) [reply example/lib/apis/general_api.dart]: SliderList

File: example/lib/main.dart [hunk 8]
```diff
@@ -2,16 +2,13 @@ import 'package:flutter/material.dart';
...
+import 'api_keys.dart';
 import 'apis/auth_apis.dart';
 import 'apis/user_apis.dart';
 import 'screens/product_screen.dart';
 
 void main() {
-  // TODO enhance example by split all APIs that related to each other in one file
-  // TODO style all buttons that call the same APIs category with the same style
-  // TODO all APIs that in the same category will be in a separated screen.
   // TODO remove the base url from the main example. [May be not because the graphConfig file]
```
Comment (MinaSamirCap) [inline example/lib/main.dart]: remove it

File: lib/common_models/common_models.dart [hunk 9]
```diff
@@ -1,91 +1,131 @@
...
+import 'package:mmd_ecommerce_fl_lib/generatedql/ratings/graphql_api.graphql.dart';
+import 'package:mmd_ecommerce_fl_lib/generatedql/user/graphql_api.graphql.dart';
+import 'package:mmd_ecommerce_fl_lib/generatedql/address/graphql_api.graphql.dart';
+import 'package:mmd_ecommerce_fl_lib/generatedql/address_asset/graphql_api.graphql.dart';
+import 'package:mmd_ecommerce_fl_lib/generatedql/orders/graphql_api.graphql.dart';
 
 /// this class to solve conflicts between the same model names of the graphql api
-class UserMyProfile {
-  final U.User user;
+class UserProfileModel {
```
Comment (MinaSamirCap) [inline lib/common_models/common_models.dart:13]: add wrapper and list models

**Review by MinaSamirCap — APPROVED** [review]

**Pull request merged**